nightfall-client/src/services/transfer.mjs

```diff
@@ -51,16 +56,35 @@ async function transfer(transferParams) {
     maticAddress: maticAddress.hex(32),
   });
 
+  const { compressedZkpPublicKey, nullifierKey } = new ZkpKeys(rootKey);
+
   const totalValueToSend = values.reduce((acc, value) => acc + value.bigInt, 0n);
-  const commitmentsInfo = await getCommitmentInfo({
+  const addedFee = maticAddress.hex(32) === ercAddress.hex(32) ? fee.bigInt : 0n;
+  const value = totalValueToSend + addedFee;
+  const feeValue = fee.bigInt - addedFee;
+
+  const txInfo = {
+    compressedZkpPublicKey,
+    ercAddress,
+    tokenId,
+    ercAddressFee: maticAddress,
+    value,
+    fee: feeValue,
+  };
+  logger.debug(`using user provided commitments: ${providedCommitments === true}`);
+  const commitments = providedCommitments || (await findUsableCommitmentsMutex(txInfo));
```

Comment:
Is there any reason why you are moving `findUsableCommitments` from getCommitmentInfo to transfer.mjs? Since we will also add a couple of new circuits, if possible I'd like to avoid code duplication. Couldn't we pass `providedCommitments` as a parameter for getCommitmentInfo and do this if inside that function?

Reply:
not sure what you mean with code duplication? does this change how the circuits work?

my general idea here was to make the commitment logic obvious in the transfer function, that's why I moved it there. but I'm open to moving it back and passing the providedCommitments if they exist.

Reply:
Well, we are also adding burn and tokenise circuits, which may use this functionality and therefore it will be duplicated code. But anyway, I think it is better to handle this logic inside the `getCommitmentInfo` function. In a future we may want to handle some more complex scenarios and it's better to handle all this logic in a single place.